lib/std/time.zig

```diff
@@ -344,6 +344,110 @@ test Timer {
     try testing.expect(timer.read() < time_1);
 }
 
+pub fn Time(precision_: comptime_int) type {
+    const multiplier: comptime_int = try std.math.powi(usize, 10, precision_);
+    return struct {
+        hour: Hour = 0,
+        minute: Minute = 0,
+        /// Allows leap seconds.
+        second: Second = 0,
+        /// Milliseconds, microseconds, or nanoseconds.
+        subsecond: Subsecond = 0,
```

Comment:
What's the justification for this additional complexity over always using nanoseconds?

Reply:
Saving bytes and allowing finer precision than nanoseconds.

It's the same argument for templating the `Year` in `Date`.

Reply:
Here are the sizes:
```zig
comptime {
    assert(@sizeOf(Time(0)) == 3);
    assert(@sizeOf(Time(3)) == 6);
    assert(@sizeOf(Time(6)) == 8);
    assert(@sizeOf(Time(9)) == 8);
}
```

If you think that calling `Time(precision)` is cumbersome I can rename `Time` -> `TimeAdvanced` and then write:
```zig
/// Time with second resolution
pub const Time = TimeAdvanced(0);
/// Time with millisecond resolution
pub const TimeMillis = TimeAdvanced(3);
/// Time with nanosecond resolution
pub const TimeNanos = TimeAdvanced(9);
```

This would be more consistent with the default `Date` and `DateTime` types.